### PR TITLE
fix: update conformance test to determine if request is full table scan

### PR DESCRIPTION
**Pull request by kevkim-codes**
The NodeJS CBT client fails the `TestReadRows_Retry_PausedScan` test with the following error message:

```
(*bigtablepb.ReadRowsRequest)(Inverse(protocmp.Transform, protocmp.Message{
  	"@type":      s"google.bigtable.v2.ReadRowsRequest",
+ 	"rows":       s"{row_ranges:[{}]}",
  	"table_name": string("projects/project/instances/instance/tables/table"),
}))
```

This is due to the fact that if the original request is a full table scan, the client request will look different on a retry (the `rows` might not exist or if it does the `rows` will be an object with an empty array because there are no `row_keys`). This was causing a false diff failure in the conformance test so we're updating this conformance test to handle this edge case. 

More info at: b/380268261

### CODE REVIEW

File: tests/readrows_test.go [hunk 1]
```diff
@@ -241,7 +241,15 @@ func TestReadRows_Retry_PausedScan(t *testing.T) {
 	origReq := <-recorder
 	retryReq := <-recorder
 	if diff := cmp.Diff(clientReq, origReq.req, protocmp.Transform(), protocmp.IgnoreEmptyMessages()); diff != "" {
-		t.Errorf("diff found (-want +got):\n%s", diff)
+		origRows := origReq.req.GetRows()
+		// Check if rows or row ranges are present in requests
+		if origRows == nil || origRows.GetRowRanges() == nil || len(origRows.GetRowRanges()) == 1 {
```
Comment (mutianf) [inline tests/readrows_test.go:246]: I think we should verify that getRowRanges is an empty range

**Review by mutianf — CHANGES_REQUESTED** [review]

File: tests/readrows_test.go (hunk 1 above)
Comment (kevkim-codes) [reply tests/readrows_test.go:246]: Can you clarify what you mean by that with an example?

Comment (mutianf) [reply tests/readrows_test.go:246]: if we're checking len(originRows.GetRowRanges) = 1, does that mean [{keyA, keyB}] will also be valid? But we want to verify it's [{}].

File: tests/readrows_test.go [hunk 2]
```diff
@@ -241,7 +241,15 @@ func TestReadRows_Retry_PausedScan(t *testing.T) {
 	origReq := <-recorder
 	retryReq := <-recorder
 	if diff := cmp.Diff(clientReq, origReq.req, protocmp.Transform(), protocmp.IgnoreEmptyMessages()); diff != "" {
-		t.Errorf("diff found (-want +got):\n%s", diff)
+		origRows := origReq.req.GetRows()
+		// Check if rows or row ranges are present in requests
```
Comment (mutianf) [inline tests/readrows_test.go:245]: Also can we update the comment to something like: "In nodejs, we add an empty row range to a full table scan request to simplify the resumption logic. This check verifies that when row ranges is present, it's also empty."

File: tests/readrows_test.go (hunk 1 above)
Comment (kevkim-codes) [reply tests/readrows_test.go:246]: Added extra check.

File: tests/readrows_test.go (hunk 2 above)
Comment (kevkim-codes) [reply tests/readrows_test.go:245]: Done.

**Review by mutianf — APPROVED** [review]
